extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -760,6 +769,31 @@ class SegmentationService extends PubSubService {
     return activeSegment;
   }
 
+  /**
+   * Gets the (last) selected segmentation for a viewport and representation type
+   *
+   * @param viewportId - The ID of the viewport to get the selected segmentation for
+   * @param segmentationRepresentationType - The type of segmentation representation to get the selected segmentation for
+   * @returns The ID of the selected segmentation, or undefined if no segmentation is/was selected for the given viewport
+   * and representation type
+   *
+   * @remarks
+   * This method retrieves the (last) selected segmentation for the specified viewport and representation type.
+   * The selected segmentation is the one that was last selected for editing operations.
+   * Returns undefined if no segmentation is selected for the specified viewport and representation type.
+   */
+  public getSelectedSegmentation({
```

Comment:
this is strange. We are holding ui state in the segmentation service. I'm not sure about this. Can you use the ui state management for it? maybe create a new `useSelectedSegmentationStore` similar to other stores we have? 
The idea is the services should not relate to ui

Reply:
I forgot all about the stores. Yes I suppose that is a good place for this.